subprojects/core/src/main/java/org/gradle/caching/BuildCacheService.java

```diff
@@ -34,6 +34,17 @@
  *     All other failures will be considered fatal and cause the Gradle build to fail.
  *     Fatal failures could include failing to read or write cache entries due to file permissions, authentication or corruption errors.
  * </p>
+ * <p>
+ *     Every build cache implementation should define a {@link org.gradle.caching.configuration.BuildCache} configuration type.
+ * </p>
+ * <p>
+ *     A build cache implementation constructor <b>must</b> conform to these requirements:
+ *     <ul>
+ *         <li>Must be public</li>
+ *         <li>Must accept an instance of the appropriate {@code BuildCache} configuration object</li>
```

Comment:
yep, you're right WRT accessibility.

I didn't want to mention that you can inject services here because I think we don't consider that completely a public feature.